staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -412,6 +412,15 @@ func (o *ApplyOptions) applyOneObject(info *resource.Info) error {
 		klog.V(4).Infof("error recording current command: %v", err)
 	}
 
+	helper := resource.NewHelper(info.Client, info.Mapping).
```

Comment:
The tri-state nature of dry-run is what makes it slightly confusing. I think we should add a lengthy comment about the tri-state nature of the dry-run. We should point out:

1) client-side dry-run means the client will prevent mutating calls (e.g. create/patch/delete).
2) server-side dry-run means the client will send the mutating calls (with server-side dry-run set, verifying the APIServer supports server-side dry-run).
3) no dry-run means the client makes all calls.

We should probably add that above the DryRunStrategy field in the ApplyOptions.

Reply:
Added a comment to cmdutil for the enums